pkg/kubectl/cmd/util/printing.go

```diff
@@ -149,19 +129,63 @@ func PrinterForCommand(cmd *cobra.Command, mapper meta.RESTMapper, typer runtime
 // object passed is non-generic, it attempts to print the object using a HumanReadablePrinter.
 // Requires that printer flags have been added to cmd (see AddPrinterFlags).
 func PrintResourceInfoForCommand(cmd *cobra.Command, info *resource.Info, f Factory, out io.Writer) error {
-	printer, generic, err := f.PrinterForCommand(cmd)
+	printer, generic, err := f.PrinterForCommand(cmd, nil)
 	if err != nil {
 		return err
 	}
 	if !generic || printer == nil {
-		printer, err = f.PrinterForMapping(cmd, nil, false)
+		printer, _, err = f.PrinterForMapping(cmd, nil, nil, false)
 		if err != nil {
 			return err
 		}
 	}
 	return printer.PrintObj(info.Object, out)
 }
 
+// extractOutputOptions parses printer specific commandline args and returns
+// printers.OutputsOptions object.
+func extractOutputOptions(cmd *cobra.Command) *printers.OutputOptions {
+	flags := cmd.Flags()
+
+	var outputFormat string
+	if flags.Lookup("output") != nil {
```

Comment:
Is `Lookup` necessary?
I think if we have flag `output`, lookup it should always be non-nil, even unspecified.